crypto/rand/rand_lib.c

```diff
@@ -25,6 +25,45 @@ static CRYPTO_RWLOCK *rand_meth_lock;
 static const RAND_METHOD *default_RAND_meth;
 static CRYPTO_ONCE rand_init = CRYPTO_ONCE_STATIC_INIT;
 
+#ifdef OPENSSL_RAND_SEED_RDTSC
+/*
+ * Since we get some randomness from the low-order bits of the
+ * high-speec clock, it can help.  But don't return a status since
+ * it's not sufficient to indicate whether or not the seeding was
+ * done.
+ */
+void rand_rdtsc(void)
+{
+    unsigned char c;
+    int i;
+
+    for (i = 0; i < 10; i++) {
+        c = (unsigned char)(OPENSSL_rdtsc() & 0xFF);
```

Comment:
I am kind of surprised NIST's Python tool ran this quickly :)  The IID testing takes an inordinate amount of time, as do a couple of the non-IID stream estimates.  The non-IID stream is used above and has both time and space problems.

What was the sampling interval used?  Was it just RDTSC in a loop or the full RAND_poll code?

Can I _guarantee_ this across all x86/x64 platforms?  No.  There are lots of platforms I don't have access to.  I have covered some older CPUs and some relatively recent ones.  I've also been conservative with the estimates so I've some confidence.

I'm also happy to analyse data collected by others and to assist with further data collection.  The more data the better.